DEV_GUIDE.md

```diff
@@ -0,0 +1,214 @@
+# Development guide
+
+This guide explains how to modify or contribute to the Forjj  project.
+Forjj is written in GO language, that needs to be compiled. This is the main difficulty, however some scripts have been written to really simplify this process.
+
+
+## Repositories
+
+To properly build Forjj you need to get the sources from github.
+
+Forjj is composed of various repositories that are available from the project main page https://github.com/forj-oss .
+- forjj --> main program, cli
+- forjj-contribs --> plugins (github, jenkins)
+- forjj-flows --> ?
+- forjj-modules --> Modules used by Forjj cli
+- forjj-repotemplates --> Templates for created project by Forjj
+- goforjj --> GO templates in order to build plugins
+
+You need to understand, which part of the project your modification will change and download the proper code repository.
+
+Advice : so far do not change the name of the default project directory because it may create troubles building the project.
+
+## Build tree
+
+Using GO, you need to have a tree similar to this inside your home directory.
+
+```
+go
+├── bin
+├── pkg
+│   └── linux_amd64
```

Comment:
Usually, when you begin, pkg is empty, or even doesn't exist. GO will create it itself.